proton-c/src/core/transport.c

```diff
@@ -1619,6 +1619,17 @@ static int pn_scan_error(pn_data_t *data, pn_condition_t *condition, const char
   return 0;
 }
 
+/*
+  This operator, copied from code for the qpid cpp broker, gives the correct
+  result when comparing sequence numbers implemented in a signed integer type.
+*/
+static bool sequence_less_than ( pn_sequence_t a, pn_sequence_t b )
```

Comment:
ie code would be

static int sequence_cmp(pn_sequence_t a, pn_sequence_t b) {
    return a-b;
}